src/modules-load/modules-load.c

```diff
@@ -150,34 +267,70 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
 
-        r = module_setup_context(&ctx);
+        /* Create a socketpair for communication with probe workers */
+        r = RET_NERRNO(socketpair(AF_UNIX, SOCK_SEQPACKET | SOCK_CLOEXEC, 0, pair));
         if (r < 0)
-                return log_error_errno(r, "Failed to initialize libkmod context: %m");
+                return log_error_errno(r, "Failed to create socket pair: %m");
+
+        /* Determine number of workers, either from env or from online CPUs */
+        num_threads = 0;
+
+        const char *e = secure_getenv("SYSTEMD_MODULES_LOAD_NUM_THREADS");
+        if (e) {
+                r = safe_atou(e, &num_threads);
+                if (r < 0)
+                        log_debug_errno(r, "Invalid value in $SYSTEMD_MODULES_LOAD_NUM_THREADS: %s", e);
```

Comment:
log messages about issues we ignore/handle gracefully should end in ", ignoring" or something, to indicate that this is not something to be overly concerned about.